### PR TITLE
Fix/lag

### LINKED ISSUE #5473: Asian chicken farm not working
# Prerequisites

- [ ] I am running the latest **alpha** version of MineColonies for my Minecraft version.
- [x] I checked the MineColonies [Wiki](https://wiki.minecolonies.ldtteam.com/) and made sure that my issue is not covered there.
- [x] I made sure that **this issue is not a duplicate of any existing issue.**
- [x] This is an issue and not a feature request. Please create all feature requests [here](https://github.com/ldtteam/minecolonies-features/issues).

### Context

- Minecraft Version: 1.15.2
- MineColonies Version:Beta 0.13.74
- Related Mods and Their Versions:

### Expected behavior

Chicken farmer got at level 3  6 chickens and he breed them any the chicken over limit got killed 

### Actual behaviour
He breed chickens but he want kill them an he got already 15 chickens at level 3 hut

### Steps to reproduce the issue

1. 
2. 
3. 
4. ... Profit

### Logs

- latest.log:  https://gist.github.com/perecnicek/b02598ff1d2f9f5c0496421f7ecf8543
- crashlog:

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #5473

Bunch of performance improvements

# Changes proposed in this pull request:
-
-
-

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAISleep.java [hunk 1]
```diff
@@ -196,18 +197,21 @@ private void findBedAndTryToSleep()
...
-                            setBedOccupied((BuildingHome) hut, true);
-                            return;
+                            final World world = citizen.world;
+                            BlockState state = world.getBlockState(pos);
+                            state = state.getBlock().getExtendedState(state, world, pos);
+                            if (state.getBlock().isIn(BlockTags.BEDS)
+                                  && !state.get(BedBlock.OCCUPIED)
+                                  && state.get(BedBlock.PART).equals(BedPart.HEAD)
+                                  && !isBedOccupied((BuildingHome) hut, pos)
+                                  && !world.getBlockState(pos.up()).getMaterial().isSolid())
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAISleep.java:209]: I also changed isAir to isSolid

**Review by Mekle001 — DISMISSED** [review]
Lots of nice perf changes. Are there before/after profiles?

**Comment by Raycoms** [comment]
I profiled it on our server. Locally it's very difficult to find a way to reproduce those.

File: src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java [hunk 2]
```diff
@@ -127,7 +128,7 @@ public void registerCivilian(final AbstractCivilianEntity entity)
             return;
         }
 
-        if (!existingCitizen.get().isAlive() || !existingCitizen.get().world.isBlockPresent(existingCitizen.get().getPosition()))
+        if (!existingCitizen.get().isAlive() || !WorldUtil.isBlockLoaded(existingCitizen.get().world, existingCitizen.get().getPosition()))
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java]: think this should be an entity block loaded check, as its for an existing citizen in a possibly not loaded chunk, we want to check if its a chunk which ticks entities

File: src/main/java/com/minecolonies/coremod/colony/managers/RaidManager.java [hunk 3]
```diff
@@ -285,7 +286,7 @@ public BlockPos calculateSpawnLocation()
 
         for (final IBuilding building : colony.getBuildingManager().getBuildings().values())
         {
-            if (colony.getWorld().isBlockPresent(building.getPosition()))
+            if (WorldUtil.isBlockLoaded(colony.getWorld(), building.getPosition()))
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/colony/managers/RaidManager.java]: entity block loaded checks in this class, we're spawning entities

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/florist/EntityAIWorkFlorist.java [hunk 4]
```diff
@@ -294,7 +295,7 @@ private BlockPos getFirstNotCompostedLand()
     {
         for (final BlockPos pos : getOwnBuilding().getPlantGround())
         {
-            if (world.isBlockPresent(pos))
+            if (WorldUtil.isBlockLoaded(world, pos))
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/florist/EntityAIWorkFlorist.java]: entity loaded check

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/herders/EntityAIWorkChickenHerder.java [hunk 5]
```diff
@@ -74,6 +74,6 @@ protected IAIState breedAnimals()
     protected IAIState butcherAnimals()
     {
         worker.getCitizenData().setVisibleStatus(FIND_CHICKEN);
-        return super.breedAnimals();
+        return super.butcherAnimals();
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/herders/EntityAIWorkChickenHerder.java:77]: oops :D

File: src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAISleep.java [hunk 6]
```diff
@@ -196,18 +197,21 @@ private void findBedAndTryToSleep()
...
                     {
-                        final World world = citizen.world;
-                        BlockState state = world.getBlockState(pos);
-                        state = state.getBlock().getExtendedState(state, world, pos);
-                        if (state.getBlock().isIn(BlockTags.BEDS)
-                              && !state.get(BedBlock.OCCUPIED)
-                              && state.get(BedBlock.PART).equals(BedPart.HEAD)
-                              && !isBedOccupied((BuildingHome) hut, pos)
-                              && world.isAirBlock(pos.up()))
+                        if (WorldUtil.isBlockLoaded(citizen.world, pos))
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAISleep.java]: entity loaded check

File: src/main/java/com/minecolonies/coremod/entity/citizen/citizenhandlers/CitizenSleepHandler.java [hunk 7]
```diff
@@ -103,7 +104,7 @@ public float getBedOrientationInDegrees()
     @Override
     public boolean trySleep(final BlockPos bedLocation)
     {
-        final BlockState state = citizen.world.isBlockPresent(bedLocation) ? citizen.world.getBlockState(bedLocation) : null;
+        final BlockState state = WorldUtil.isBlockLoaded(citizen.world, bedLocation) ? citizen.world.getBlockState(bedLocation) : null;
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/entity/citizen/citizenhandlers/CitizenSleepHandler.java]: entity loaded

File: src/api/java/com/minecolonies/api/util/constant/CitizenConstants.java [hunk 8]
```diff
@@ -115,6 +115,12 @@
      * 20 ticks or also: once a second.
      */
     public static final int    TICKS_20                   = 20;
+
+    /**
+     * 20 0ticks or also: once every 10 seconds.
```
Comment (Asherslab) [inline src/api/java/com/minecolonies/api/util/constant/CitizenConstants.java]: typo

**Review by Asherslab — DISMISSED** [review]
looks good to me

**Review by someaddons — APPROVED** [review]

**Review by Mekle001 — APPROVED** [review]

**Review by Asherslab — APPROVED** [review]

**Pull request merged**